avro-serializer/src/main/java/io/confluent/kafka/serializers/AbstractKafkaAvroDeserializer.java

```diff
@@ -270,16 +276,20 @@ protected GenericContainerWithVersion deserializeWithSchemaAndVersion(
     // explicit from the Connector).
     DeserializationContext context = new DeserializationContext(topic, isKey, headers, payload);
     AvroSchema schema = context.schemaForDeserialize();
-    Object result = context.read(schema,
-        specificAvroReaderSchema != null ? new AvroSchema(specificAvroReaderSchema) : null);
+    AvroSchema readerAvroSchema = readerSchema != null
```

Comment:
nit: I personally think nested ternary expressions is not super readable, compared to old fashioned if-else